tests/pypi/extension/extension_tests.bzl

```diff
@@ -182,6 +185,46 @@ def _test_simple(env):
 
 _tests.append(_test_simple)
 
+def _test_simple_platform_pip_args(env):
+    pypi = _parse_modules(
+        env,
+        module_ctx = _mock_mctx(
+            _mod(
+                name = "rules_python",
+                parse = [
+                    _parse(
+                        hub_name = "pypi",
+                        python_version = "3.15",
+                        requirements_darwin = "darwin.txt",
+                        extra_pip_args_by_platform = {
+                            "macosx_*": ["--only-binary=simple"],
+                        },
+                    ),
+                ],
+            ),
+            os_name = "mac os x",
+            os_arch = "arch64",
+            read = lambda x: {
+                "darwin.txt": "simple==0.0.2 --hash=sha256:deadb00f",
```

Comment:
Could you please add extra files so that we can see that there are differences on each platform?